xray_vision/mpl_plotting/speckle.py

```diff
@@ -0,0 +1,318 @@
+    color_cycle : str
+        Matplotlib string name of colormap (see matplotlib.pyplot.colormaps()
+        for a list of valid colormaps on your machine)
+    
+    Returns
+    -------
+    artists : pd.DataFrame
+        Pandas DataFrame whose column/row names match the input `dataframe`
+    
+    Examples
+    --------
+    
+    """
+    if cmap is None:
+        # TODO don't use viridis in production, yet...
+        if 'viridis' in plt.colormaps():
+            cmap = 'viridis'
+        else:
+            cmap = 'rainbow'
+    cmap = plt.get_cmap(cmap)
+
+    ax[-1].set_xlabel(xlabel)
+    # capture the artists in a nested dictionary
+    artists = {col_name: {} for col_name in dataframe}
+    # compute the offset for each column
+    offsets = []
+    cur = 0
+    prev = 0
+    # determine how far to offset each data set
+    for data in dataframe.ix[0]:
+        cur = prev + len(data)
+        offsets.append((prev, cur))
+        prev = cur
+    # loop over the rows of the dataframe
+    for idx, row_label in enumerate(dataframe.index):
+        # do some axes housekeeping
+        ax[idx].set_ylabel(ylabel)
+        ax[idx].set_title(title + ' for %s' % row_label)
+        row = dataframe.ix[row_label]
+        # loop over the columns of the dataframe, creating each line plot
+        # one at a time
+        for idx2, (column_name, color_idx) in enumerate(zip(
+            dataframe, np.arange(0, 1, 1/len(row)))):
+            x = range(*offsets[idx2])
+            y = row.ix[column_name]
+            art, = ax[idx].plot(x, y, label=column_name, color=cmap(color_idx))
+            # store the artists in a nested dictionary
+            artists[column_name][row_label] = art
+        # enable the legend for each plot after all data has been added
+        ax[idx].legend()
+    return pd.DataFrame(artists)
+
+
+def combine_intensity_plotter(ax, combine_intensity,
+                              title="Mean Intensities - All Image Sets",
+                              xlabel="Frames", ylabel="Mean Intensity",
+                              labels=None):
+    """
+    This will plot the combine intensities for all image sets
+
+    Parameters
+    ----------
+    ax : Axes
+        The matplotlib.axes.Axes object to add the roi data to
+    combine_intensity : list
+        List of intensities for each ROI. Each element in the list should be
+        a 1-D array where the x-axis is understood to be frame number
+    title : str, optional
+        title of the plot
+    x_label : str, optional
+        x axis label
+    y_label : str, optional
+        y axis label
+    labels : list, optional
+        Names for each ROI data set. If a list is provided, it should be the
+        same length as the `combine_intensity` list. If a list is not provided,
+        the default will be 'ROI #' where # is the index of the dataset in
+        `combine_intensity`+1
+    """
+    num_rois = len(combine_intensity)
+    if labels is None:
```

Comment:
I think we need "The labels for each element of the `mean_intensity` list" before combining it, Therefore, it shouldn't be optional.
Before combining it is better to check all the mean intensities from the different image sets have same label list(index list)
Otherwise it can combine different  roi's?